### PR TITLE
Milestone 2 components

### PR DESCRIPTION
**Pull request by m15e**
- [ ] created components, reducers, actions and redux store according to milestone-2 specifications

### CODE REVIEW

File: src/App.js
Comment (shloch) [inline src/App.js]: Relocate this component to `src/components/` folder

File: src/components/Book.js
Comment (shloch) [inline src/components/Book.js]: This component is expected to accept `Book` object as prop. Meaning you'll have to pass the whole object instead of the individual properties
[[ see requirements](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/Book/AccceptBookObjectAsProp.png) ]

File: src/components/BooksList.js
Comment (shloch) [inline src/components/BooksList.js]: pass the whole object

Comment (shloch) [reply src/components/BooksList.js]: Relocate this component into the `src/containers/` folder,  [ see requirements ]

File: src/components/BooksForm.js
Comment (shloch) [inline src/components/BooksForm.js]: relocate this component into `src/containers/` folder

**Review by shloch — CHANGES_REQUESTED** [review]
## Hello @m15e 

Great job. All the logic needed to build this project at the level of this milestone is well done by you. I however have a series of small things to let you know to meet ALL the requirements.

Given your enormous hard work, I trust that you will finish this in no time
### Waiting for a final review then [ see comments below ].
HAPPY CODING

**Comment by m15e** [comment]
To the reviewing TSE: I've made all the changes that the previous TSE requested but could not comply with the second request to pass in the entire object as a prop, as it caused trouble with the linter. I've tried multiple ways to do this but couldn't get the linter to pass without disabling the no prop spreading rule. I'd appreciate it if you could either approve the project or provide guidance on how to solve this without eslint popping off. Thanks and have a good day :)

**Comment by shloch** [comment]
File for a new review, also include the linting error(screenshot), a TSE will pick it up

File: src/components/Book.js
Comment (Forison) [inline src/components/Book.js:4]: **[optional]**
You could also handle it this way `const Book = ({ id, title, category }) => {`

**Review by Forison — APPROVED** [review]
# Status::Approved✔️

#### Good job on addressing the issues raised in the issues raised in the previous review, go ahead and merge the branch and proceed to the next milestone🤗.
> Kindly create a zendesk ticket about the error you spoke about in the pull request, It is not too clear what said caused the error.

#### Happy coding👨‍💻

Addo Forison

**Pull request merged**
